tests/api_fastapi/core_api/routes/public/test_connections.py

```diff
@@ -235,9 +235,7 @@ def test_post_should_respond_already_exist(self, test_client, body):
         # Another request
         response = test_client.post("/public/connections/", json=body)
         assert response.status_code == 409
-        assert response.json() == {
-            "detail": f"Connection with connection_id: `{TEST_CONN_ID}` already exists",
-        }
+        assert response.json() == {"detail": "Unique constraint violation"}
```

Comment:
Can we subclass the global exception handler to add personalized messages? 'Unique constraint violation' is quite broad

Reply:
It is possible to include error details (e.g., `str(e)`) to provide a more descriptive message. However, I can't think of a way to include custom error messages specific to each router, as the error handler is registered at the `app` level rather than the router level.

Reply:
I don't like the refactor if we can't customize the response detail.

Reply:
Let me research if there is a way to register custom error message for specific routers.

Reply:
What type of error would you like to display @ephraimbuddy ?

Do we expect the name of the Model, of the db table, or just one message for all but just a more explicit one, or the message coming from the database ?

Reply:
After conducting some research, I have come up with a solution to add custom error message handlers for specific endpoints. If a custom error handler is provided, it will take precedence and be used before the default global error handler.

The usage would look like this:
```python
from airflow.api_fastapi.common.exceptions import register_unique_constraint_error_handler, BaseCustomErrorHandler


@connections_router.post(
    "",
    status_code=status.HTTP_201_CREATED,
    responses=create_openapi_http_exception_doc(
        [status.HTTP_409_CONFLICT]
    ),  # handled by global exception handler
)
def post_connection():
# ....

class _ConnectionUniqueConstraintErrorHandler(BaseCustomErrorHandler):

    def exception_handler(self, request_schema: ConnectionBody):
        raise HTTPException(
            status_code=status.HTTP_409_CONFLICT,
            detail=f"Connection with connection_id: `{request_schema.connection_id}` already exists",
        )

register_unique_constraint_error_handler(post_connection, _ConnectionUniqueConstraintErrorHandler(ConnectionBody))
```
Not sure is this clean enough for providing verbose error message for specific endpoints, WDYT ?

Reply:
Maybe returning the `orig` is enough ? There is enough information for the user to understand what's going on, but that's still a pretty technical message for the user...

Reply:
Other idea, create an issue and solve that in a successive PR. This one goal is just to make things concitent accross the different endpoints.

Reply:
> What type of error would you like to display @ephraimbuddy ?

The previous error message is more detailed than the generalized one. I think we should continue with the current implementation that tells the user what is wrong instead of a general one that would be confusing in some endpoints. I don't have a strong opinion on this.

Reply:
I will try a regex or string-matching approach, as the error detail format for the same database driver should remain consistent. If this works, it could provide generic error details on the global error handler side.

Reply:
Maybe just `orig` and `statement` is fine. And for tests we just assert that those keys are present in the response.